src/control/cmd/daos/filesystem.go

```diff
@@ -262,27 +262,69 @@ func (cmd *fsGetAttrCmd) Execute(_ []string) error {
 	defer cleanup()
 
 	var attrs C.dfs_obj_info_t
-	if err := dfsError(C.fs_dfs_get_attr_hdlr(ap, &attrs)); err != nil {
+	var cmode C.mode_t
+	if err := dfsError(C.fs_dfs_get_attr_hdlr(ap, &attrs, &cmode)); err != nil {
 		return errors.Wrapf(err, "%s failed", fsOpString((ap.fs_op)))
 	}
 
 	var oclassName [16]C.char
 	C.daos_oclass_id2name(attrs.doi_oclass_id, &oclassName[0])
 
+	var diroclassName [16]C.char
+	var fileoclassName [16]C.char
+	if C.mode_is_dir(cmode) {
+		C.daos_oclass_id2name(attrs.doi_dir_oclass_id, &diroclassName[0])
+		C.daos_oclass_id2name(attrs.doi_file_oclass_id, &fileoclassName[0])
+	}
+
 	if cmd.JSONOutputEnabled() {
-		jsonAttrs := &struct {
-			ObjClass  string `json:"oclass"`
-			ChunkSize uint64 `json:"chunk_size"`
-		}{
-			ObjClass:  C.GoString(&oclassName[0]),
-			ChunkSize: uint64(attrs.doi_chunk_size),
+		if C.mode_is_dir(cmode) {
```

Comment:
nit: I think it's tidier to return early so that you can reduce nesting:
```
	if C.mode_is_dir(cmode) {
	        if !cmd.JSONOutputEnabled() {
                	cmd.Infof("Object Class = %s", C.GoString(&oclassName[0]))
                        cmd.Infof("Directory Creation Object Class = %s", C.GoString(&diroclassName[0]))
          		cmd.Infof("File Creation Object Class = %s", C.GoString(&fileoclassName[0]))
          		cmd.Infof("File Creation Chunk Size = %d", attrs.doi_chunk_size)
           	        return nil
          	}

		return cmd.OutputJSON(struct {
			ObjAttr struct {
				ObjClass string `json:"oclass"`
			} `json:"Object"`
			DirAttr struct {
				DirObjClass  string `json:"dir_oclass"`
				FileObjClass string `json:"file_oclass"`
				ChunkSize    uint64 `json:"chunk_size"`
			} `json:"Directory"`
		}{
			ObjAttr: struct {
				ObjClass string `json:"oclass"`
			}{
				ObjClass: C.GoString(&oclassName[0]),
			},
			DirAttr: struct {
				DirObjClass  string `json:"dir_oclass"`
				FileObjClass string `json:"file_oclass"`
				ChunkSize    uint64 `json:"chunk_size"`
			}{
				FileObjClass: C.GoString(&diroclassName[0]),
				DirObjClass:  C.GoString(&fileoclassName[0]),
				ChunkSize:    uint64(attrs.doi_chunk_size),
			},
		}, nil)
	}

        if !cmd.JSONOutputEnabled() {
               	cmd.Infof("Object Class = %s", C.GoString(&oclassName[0]))
		cmd.Infof("Object Chunk Size = %d", attrs.doi_chunk_size)
	        return nil
	}
	
	return cmd.OutputJSON(&struct {
		ObjClass  string `json:"oclass"`
		ChunkSize uint64 `json:"chunk_size"`
	}{
		ObjClass:  C.GoString(&oclassName[0]),
		ChunkSize: uint64(attrs.doi_chunk_size),
	}, nil)
}
```